libmobilecoin/src/transaction.rs

```diff
@@ -147,12 +166,14 @@ pub extern "C" fn mc_tx_out_get_value(
     out_error: FfiOptMutPtr<FfiOptOwnedPtr<McError>>,
 ) -> bool {
     ffi_boundary_with_error(out_error, || {
-        let amount = Amount::try_from_ffi(&tx_out_amount)?;
         let tx_out_public_key = RistrettoPublic::try_from_ffi(&tx_out_public_key)?;
         let view_private_key = RistrettoPrivate::try_from_ffi(&view_private_key)
             .expect("view_private_key is not a valid RistrettoPrivate");
 
         let shared_secret = get_tx_out_shared_secret(&view_private_key, &tx_out_public_key);
+        let value =
+            (tx_out_amount.masked_value as u64) ^ get_value_mask(&shared_secret);
+        let amount: Amount = Amount::new(value, &shared_secret)?;
         let (val, _blinding) = amount.get_value(&shared_secret)?;
```

Comment:
Btw I think `value == val` here.

Reply:
Ya I agree, so maybe it could be optimized. Do you know what the `_blinding` field represents?

Reply:
It is the amount's Pedersen commitment's blinding factor.

Reply:
updated to pass out the `value` instead of creating an `Amount` object